### PR TITLE
Common: Add errno_str which gives string for errno

### LINKED ISSUE #26: Feature: errno_str(int errno)
get string description for errno.

### PR DESCRIPTION
**Pull request by yuandong1222**
Close #26

Signed-off-by: Dong Yuan <yuandong1222@gmail.com>

### CODE REVIEW

File: src/common/errno.c [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#include "errno.h"
+
+extern const char* errno_str(int errno) {
```
Comment (ketor) [inline src/common/errno.c]: Why use extern here?

**Comment by ketor** [comment]
LGTM

File: src/common/errno.c (hunk 1 above)
Comment (yuandong1222) [inline src/common/errno.c]: I am not sure whether this keyword is necessary here.

Comment (ketor) [inline src/common/errno.c]: Not necessary. Only used for declare a function before calling it.

Comment (yuandong1222) [inline src/common/errno.c]: OK, I will remove it and all some cases.

**Pull request merged**